components/director/internal/domain/application/resolver.go

```diff
@@ -291,6 +291,20 @@ func (r *Resolver) Application(ctx context.Context, id string) (*graphql.Applica
 	return r.appConverter.ToGraphQL(app), nil
 }
 
+// ApplicationBySystemNumber missing godoc
+func (r *Resolver) ApplicationBySystemNumber(ctx context.Context, systemNumber string) (*graphql.Application, error) {
```

Comment:
An e2e test can be added for the resolver in the application_api_test.go file. You need to add a fixture for the request in this file kyma-incubator/compass/tests/pkg/fixtures/application_requests.go like so: 
````
func FixGetApplicationBySystemNumberRequest(systemNumber string) *gcli.Request {
	return gcli.NewRequest(
		fmt.Sprintf(`query {
			result: applicationBySystemNumber(systemNumber: "%s") {
					%s
				}
			}`, systemNumber, testctx.Tc.GQLFieldsProvider.ForApplication()))
}
````



In the e2e test you can do a:
`saveExample(t, getApplicationBySystemNumberRequest.Query(), "get application by systemNumber")`
See other places for reference. 
At the end do: `make generate-examples`  in the `tests/director` folder where e2e tests are to generate an example file for the GraphQL request and commit it.